src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,425 @@
+            }
+        }
+
+        return $translatorBag;
+    }
+
+    public function delete(TranslatorBagInterface $translatorBag): void
+    {
+        $filesMap = $this->getFilesMap();
+
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            if ($catalogue->getLocale() !== $this->defaultLocale) {
+                continue;
+            }
+
+            foreach ($catalogue->getDomains() as $domain) {
+                $fileId = $this->getFileIdByDomain($filesMap, $domain);
+
+                if (!$fileId) {
+                    continue;
+                }
+
+                $stringsMap = $this->mapStrings($fileId);
+
+                foreach ($catalogue->all() as $messages) {
+                    foreach ($messages as $id => $message) {
+                        if (!isset($stringsMap[$id])) {
+                            continue;
+                        }
+
+                        $this->deleteString($stringsMap[$id]);
+                    }
+                }
+            }
+        }
+    }
+
+    private function getFilesMap(): array
+    {
+        $result = [];
+
+        $fileList = $this->getFileList();
+
+        foreach ($fileList as $file) {
+            $result[$file['data']['name']] = $file['data']['id'];
+        }
+
+        return $result;
+    }
+
+    private function getFileIdByDomain(array $filesMap, string $domain): ?int
+    {
+        return $filesMap[sprintf('%s.%s', $domain, 'xlf')] ?? null;
+    }
+
+    private function mapStrings(int $fileId): array
+    {
+        $result = [];
+
+        $limit = 500;
+        $offset = 0;
+
+        do {
+            $strings = $this->listStrings($fileId, $limit, $offset);
+
+            foreach ($strings as $string) {
+                $result[$string['data']['text']] = $string['data']['id'];
+            }
+
+            $offset += $limit;
+        } while (\count($strings) > 0);
+
+        return $result;
+    }
+
+    private function addFile(string $domain, string $content): array
+    {
+        $storageId = $this->addStorage($domain, $content);
+
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.files.getMany (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.files.getMany (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('POST', sprintf('https://%s/projects/%s/files', $this->endpoint, $this->projectId), [
+            'headers' => [
+                'Content-Type' => 'application/json',
+            ],
+            'body' => json_encode([
+                'storageId' => $storageId,
+                'name' => sprintf('%s.%s', $domain, 'xlf'),
+            ]),
+        ]);
+
+        $responseContent = $response->getContent();
+
+        if (201 !== $response->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to create a File in Crowdin for domain "%s": "%s".', $domain, $responseContent), $response);
```

Comment:
You should use the Logger here, because `addFile` is called in foreach loops that we don't want to break.